feat: hide filter by building + mobile drawer

cnk2024 opened the pull request:
- Hid filter by building for mobile view
- Replaced details/menu drawer with bottom sheet for mobile

Arom1a committed:
chore: change bar imdicator to dot

cnk2024 committed:
Merge branch 'card-refactor' of https://github.com/ScottyLabs/cmueats into card-refactor

cnk2024 committed:
Header adjustments

cnk2024 committed:
Merge branch 'eatery-card-refactor' into ui/refactored-mobile-redesign

cnk2024 committed:
Finished header, shrunk cards

WIP: More menu / pinning in header

cnk2024 committed:
Merge branch 'eatery-card-refactor' into ui/refactored-mobile-redesign

cnk2024 committed:
Refactored Eatery Card Dropdown

cnk2024 committed:
Added dropdown to mobile eatery card header

cnk2024 committed:
Minor CSS changes

cnk2024 committed:
Minor css

cnk2024 committed:
Revert Added dropdown to mobile eatery card header"

cnk2024 committed:
Revert "Refactored Eatery Card Dropdown"

This reverts commit 60e2792d8e2fd80b881d8960319188e039fdce54.

cnk2024 committed:
Pin/Hide on header

cnk2024 committed:
Merge eatery-card-refactor, disable drawer

cnk2024 committed:
Merge branch 'staging' into ui/refactored-mobile-redesign-staged

cnk2024 committed:
Consistency with desktop + styling fixes

- Made some sizes smaller
- Changed some spacing

cnk2024 committed:
Condensed padding further

cnk2024 committed:
Added back pinning for mobile only

cnk2024 committed:
Fixed small bug with 600 size

Search bar width fixed

cirex-web commented on src/index.css:
```diff
@@ -20,6 +20,7 @@
     --text-primary: white;
     --text-greeting: white;
     --text-muted: #d4d4d8;
+    --black-400: #b4bbc0;
```
same color as above?

cirex-web commented:
thanks, will review soon

cnk2024 committed:
Removed repeat color

cnk2024 committed:
Fix type error in ListPage

Should have no changes in behavior

cnk2024 committed:
Added bottom sheet component (drawer for mobile)

cnk2024 committed:
Mobile Drawer (WIP, no animations or custom spacing)

cnk2024 committed:
Added more snap points to bottom sheet

cnk2024 committed:
Spacing/sizing edits for mobile drawer

cnk2024 committed:
Expanded handle bounding box

cnk2024 committed:
Bottom sheet adjustments

- Easier to manipulate/dismiss bounding box (increased hitbox of handle + dismiss drawer when clicking out of it)
- Dimming effect on rest of the screen

cnk2024 committed:
Added flick gestures

Flick down to immediately dismiss

cnk2024 committed:
Added entry animation to bottom sheet

cnk2024 committed:
Entry/exit animation added to bottom sheet

slight bug: display flickers

cnk2024 committed:
Fixed exit/entry animation for bottom sheet

cnk2024 committed:
Merge branch 'staging' into ui/refactored-mobile-redesign-staged

cnk2024 committed:
Minor visual updates

- Fixed previous merge
- Added fade transition to bottom sheet dimming
- Adjusted size/spacing
- Brought back longer location

cnk2024 committed:
Prettier Lint formatting

cnk2024 committed:
Styling fixes (lint), bottom sheet bug fix

Fixed minor bug with bottom sheet entry

cirex-web commented on src/pages/ListPage.css:
```diff
@@ -136,3 +138,53 @@
+        display: flex;
+        align-items: center;
+        gap: 7px;
+        padding-top: 1.1rem;
```
oh yea we'll be using `px` for everything now btw, just for consistency

cirex-web committed:
style: minor mobile style tweaks

cirex-web commented on src/components/EateryCardGrid.tsx:
```diff
@@ -93,11 +93,15 @@ export default function EateryCardGrid({
                 <Info size={16} aria-hidden="true" />
-                <span>Tap or click on the cards to see more information!</span>
+                <span>
+                    {isMobile ? 'Tap on a card for more details' : 'Tap or click on the cards to see more information!'}
```
nice

cirex-web committed:
style: minor tweaks

cirex-web commented on src/pages/ListPage.css:
```diff
@@ -136,3 +138,53 @@
+
+    .list-controls-layout {
+        margin: 0px;
+        padding:;
```
mrrp

cnk2024 committed:
Mobile compact card style tweaks (#753)

gave the card header a bit more room, but it stil fits 5 cards on my
screen, like the original design

cnk2024 committed:
isMobile context

Checking window for mobile size is now handled in a context

cnk2024 committed:
Fixed minor warnings in isMobile context

cnk2024 committed:
Shortened delay on bottom sheet

cnk2024 committed:
Minor styling changes

cnk2024 committed:
Bottom sheet scrolling + expansion

- Scroll up to expand bottom sheet
- Scrolling only enabled when fully expanded

cnk2024 committed:
Merge branch 'staging' into ui/refactored-mobile-redesign-staged

cnk2024 committed:
Removed in-between snap points and locked scrolling for bottom sheet

cnk2024 committed:
Fixed mobile reviews

cnk2024 committed:
Fixed scroll lock for bottom sheet + exit animation

cnk2024 committed:
Fixed Bottom Sheet exit animation and flickering

cnk2024 committed:
Moved Navbar to the front

cnk2024 committed:
Made bottom sheet scroll lock more strict

cnk2024 committed:
Merge branch 'staging' into ui/refactored-mobile-redesign-staged

cnk2024 committed:
Removed redundant line

cnk2024 committed:
Much more reliable mobile scroll lock when drawer is open

cnk2024 committed:
Prevented drawer overscroll on mobile

cnk2024 committed:
Can dismiss whole mobile drawer easier

If the content inside the bottom sheet is at the top, scrolling down dismisses the sheet

cnk2024 committed:
Merge branch 'staging' into ui/refactored-mobile-redesign-staged

cnk2024 committed:
fix: smooth exit transition on mobile drawer

Fixed bug where bottom sheet disappears when halfway dismissed

cnk2024 committed:
style: bottom sheet readability

- Made all functions to constants for consistency
- Removed redundant requestAnimationFrame
- functions to eliminate casing inside onMouseDown and onTouchStart

cnk2024 committed:
fix: smoother entry/exit animation for mobile drawer

cnk2024 committed:
fix: uncommented accidentally commented line

Removed for testing, forgot to add that back

cnk2024 committed:
style: remove large fragment

Removed fragment from flick control implementation (nonfunctional)

cnk2024 committed:
fix: mobile drawer dismissed easier

Allows the mobile drawer to be dismissed easier when sliding down

cnk2024 committed:
style: made variable name clearer

Made variable name more specific to its function

cnk2024 committed:
style: fixed lint errors

cnk2024 committed:
fix: removed fragments from compressed mobile view

Removed unnecessary changes to mobile UI

cnk2024 committed:
style: removed more leftover fragments

cnk2024 committed:
Merge branch 'staging' into ui/refactored-mobile-redesign-staged

cnk2024 committed:
fix: locked scrolling in bottom sheet content

cnk2024 committed:
fix: windowHeight no longer stale

cnk2024 committed:
fix: 'scroll' removed from correct element

cnk2024 committed:
style: appease linter

cnk2024 committed:
fix: clear timeouts properly

cnk2024 committed:
fix: fix bug where scrolling prevented when switching to map

cirex-web commented on src/components/DrawerHeader.module.css:
```diff
@@ -74,3 +74,24 @@
+
+@media (max-width: 600px) {
+    .drawer-header-container {
+        padding: 0px 27px 20px;
```
why the increase in horizontal margin to 27px?

cnk2024 replied:
Oops, looks like that margin was hard set to 27px when I made the branch and it didn't get updated when I merged staging with it later. Fixed by changing it to var(--drawer-x-padding) to match the desktop version

cirex-web commented on src/components/DrawerHeader.module.css:
```diff
@@ -74,3 +74,24 @@
+    }
+
+    .location-link {
+        font-size: 25px;
```
and an increase in 1px in the font size?

cnk2024 replied:
I think that was a fragment left over from when I was trying to resize things for mobile to match the figma specifications. Removed this in latest commit

cirex-web commented on src/components/EateryCard.module.css:
```diff
@@ -88,3 +88,9 @@
+@media (max-width: 600px) {
+    .desktop {
+        display: none;
+    }
```
is this used anywhere?

cnk2024 replied:
Another fragment from when we were debating pins on the header vs. dropdown I think(?), it isn't used so I removed it in the latest commit

cirex-web commented on src/contexts/DrawerAPIContext.tsx:
```diff
@@ -21,10 +21,6 @@ export function DrawerAPIContextProvider({ children }: { children: React.ReactNo
-        // prevent the card list from being scrolled on mobile, kind of works...
-        document.body.style.overflow = drawerConceptId === null ? 'visible' : 'hidden';
-    }, [drawerConceptId]);
     const drawerContextValue = useMemo(
```
hmm might we still need this when the window width makes the drawer full-screen but not yet a BottomSheet? (or should we just transition directly to BottomSheet a bit earlier? (aka at a larger width threshold)

<img width="221" height="260" alt="Image" src="https://github.com/user-attachments/assets/cc14fb27-cda2-4323-b672-915c9166378f" />

cnk2024 replied:
I have custom logic for mobile scroll locking, so I assumed that it would be redundant to do this scroll lock since the comment seemed to imply it was only for mobile. I restored it in the latest commit since its also important for desktop.

cirex-web commented on src/pages/ListPage.tsx:
```diff
@@ -62,7 +65,7 @@ function ListBox({
                 <div className="list-controls-layout">
                     <SearchBar searchQuery={searchQuery} setSearchQuery={setSearchQuery} />
-                    <SelectLocation {...{ setLocationFilterQuery, locations }} />
+                    {!isMobile && <SelectLocation {...{ setLocationFilterQuery, locations }} />}
```
fair

cirex-web commented on src/components/BottomSheet.tsx:
```diff
@@ -0,0 +1,264 @@
+        document.removeEventListener('touchmove', preventScroll);
+        document.removeEventListener('wheel', preventScroll);
+        document.documentElement.classList.remove('scroll-lock');
+    }, [preventScroll]);
```
why not have a single useEffect with [active] as its dependency array? If active is true, prevent scrolling, and delete the event listeners in the cleanup returned callback. (also, might it be possible to lock scrolling with css instead? naively I would try just adding a transparent fullscreen div behind the BottomSheet. Or maybe disabling pointer events? idk)

cnk2024 replied:
The lockScroll and unlockScroll are already used in a single useEffect with [active] as its dependency array. They're just helpers to keep the code readable. Think this is the best way I could come up with to lock scrolling since it changes properties that css would control anyway (overflow). There's already a translucent fullscreen div behind the BottomSheet for a visual dim effect, but unfortunately React just lets you scroll "through" it since there's no text or scrollable content. I've tried disabling pointer events before to little success since it doesn't block user scrolling any more effectively than just setting overflow to hidden and has some issues with preventing other inputs.

The current solution is to have an initial scroll lock that just sets the document body's overflow to lock/unlock scrolling just like DrawerAPIContext does, but since it is notoriously finicky on mobile I have an additional lock that blocks user input if they try to scroll an element outside the bottom sheet. This has been the most consistent fix to the scroll lock so far since the main issue with the overflow method is that it doesn't work all the time for some reason.

cirex-web replied:
ok ye makes sense

cirex-web commented on src/components/BottomSheet.tsx:
```diff
@@ -0,0 +1,264 @@
+
+            const endTime = Date.now();
+            const dt = endTime - dragStartTime.current;
+            const dy = clientY - dragStartY.current;
```
consider using `performance.now()` instead? It's a bit more precise than `Date.now()` and is intended for things like this
https://developer.chrome.com/blog/when-milliseconds-are-not-enough-performance-now?hl=en

cnk2024 replied:
Now using performance.now() in newest commit

cnk2024 committed:
fix: stuck scrolling

Fixed bug where cards cannot be scrolled after opening the bottom sheet

cnk2024 committed:
fix: reviewed changes

- Removed mobile resizing fragments (unnecessarily changed css for mobile)
- useCallback hook for several functions (startDrag, onSheetScroll)
- Restored scroll lock for non-mobile drawer
- Bottom sheet uses performance.now() instead of Date.now() for flick detection
- Synchronized activeRef with active

cirex-web commented:
thanks for the changes!
There seems to be a bug if you scroll down on desktop and then drag down, the drawer panel jumps

https://github.com/user-attachments/assets/97059acb-bcef-4339-b60b-f61d35f338b4

cnk2024 committed:
fix: bottom sheet "jumping" and scroll locking after going to map

cnk2024 committed:
Merge branch 'staging' into ui/refactored-mobile-redesign-staged

Arom1a commented on src/components/BottomSheet.tsx:
```diff
@@ -0,0 +1,268 @@
+                        if (e.key === 'Escape') hide();
+                    }}
+                    tabIndex={0}
+                    role="button"
```
just add `// oxlint-disable-next-line jsx_a11y/prefer-tag-over-role` to disable this lint rule

Arom1a commented on src/components/BottomSheet.module.css:
```diff
@@ -0,0 +1,64 @@
+.bottomSheet {
+    z-index: 998;
```
why is it 998 instead of 999 lol

Arom1a replied:
does not matter tho

cirex-web commented on src/components/BottomSheet.tsx:
```diff
@@ -0,0 +1,268 @@
+};
+
+export default function BottomSheet({ children, active, onHide }: BottomSheetProps) {
+    const DELAY = 100;
```
rename to a slightly more intuitive name?

cirex-web commented on src/components/BottomSheet.tsx:
```diff
@@ -0,0 +1,268 @@
+
+    const [windowHeight, setWindowHeight] = useState<number>(window.innerHeight);
+    const [FULL, setFULL] = useState<number>(windowHeight * 0.15);
+    const [HIDDEN, setHIDDEN] = useState<number>(windowHeight);
```
these two don't have to be states I think?

cnk2024 committed:
style: appease linter

Arom1a commented on src/contexts/IsMobileContext.tsx:
```diff
@@ -0,0 +1,24 @@
+
+const isMobileContext = createContext<boolean | undefined>(undefined);
+
+export function IsMobileContextProvider({ children }: { children: React.ReactNode }) {
```
I don't see a need to add a context for this. We could use a hook instead, e.g.:

```ts
function useIsMobile(breakpoint = 600) {
      const [isMobile, setIsMobile] = useState(() => window.innerWidth <= breakpoint);

      useEffect(() => {
          const mediaQuery = window.matchMedia(`(max-width: ${breakpoint}px)`);

          const update = () => setIsMobile(mediaQuery.matches);
          update();

          mediaQuery.addEventListener('change', update);
          return () => mediaQuery.removeEventListener('change', update);
      }, [breakpoint]);

      return isMobile;
  }
```

cnk2024 committed:
style: format

cnk2024 committed:
style: clarifying comments, oxlint disable

cnk2024 committed:
style: refactored dragging logic for bottom sheet

cnk2024 committed:
style: more rigorous check for if bottom sheet is at top

cnk2024 committed:
style: annotations, minor stylistic changes

cirex-web committed:
chore: format

cirex-web reviewed (approved):
makes sense, thanks!
